pulp_rpm/app/tasks/copy.py

```diff
@@ -93,13 +93,15 @@ def find_children_of_content(content, src_repo_version):
     for packagegroup in packagegroups.iterator():
         group_package_names = [pkg["name"] for pkg in packagegroup.packages]
         for pkg in group_package_names:
-            packages_by_name = [
-                pkg
-                for pkg in Package.objects.with_age().filter(
-                    name=pkg, pk__in=src_repo_version.content
-                )
-                if pkg.age == 1
-            ]
+            packages_by_name = Package.objects.filter(name=pkg, pk__in=content)
+            if not packages_by_name.exists():
+                packages_by_name = [
+                    pkg
+                    for pkg in Package.objects.with_age().filter(
+                        name=pkg, pk__in=src_repo_version.content
+                    )
+                    if pkg.age == 1
+                ]
```

Comment:
@ggainey Do you think we should also account for when the destination repository already contains a particular package?

Although I would be content to leave it here.

Reply:
My use case would want latest versions left out whether or not Katello is copying everything in the same operation or not, and I am insufficiently well-versed to speak confidently about whether Katello always behaves the way it has in my tests.  On the other hand, maybe for other use cases, copying a package group to a repository that already has it means you want the latest versions?

In case y'all do end up wanting it, I've pushed a second commit to check the destination, with the caveat that I've only tested my Katello paths.

Reply:
My brain is mildly mush, let me state what I think we're doing here and you two can correct me as needed, pliz.  

Current behavior is "get all packages-by-name from package-groups and add latest-version, whether it's already been selected for copy or not." And this means that if we've already carefully selected a non-latest version of a package, it gets overridden by "latest" if it's in a packagegroup. As a side(?)-effect, destination ends up with 'latest' of all packages in a packagegroup, when we're done.

First-commit (9d08a5) is "get all packages-by-name where said name isn't already in the selected-for-copy group," to fix the "overrides current selection". 

@dralley's thought, and @quartsize 's 2nd commit (6dac5e), extends that to "and the package isn't in the destination already". The diff between this and 9d08a5 is that if dest has old-pkg-version, and we haven't already selected it, 9d08a5 would put latest-version into dest, and 6dac5e , won't.

So, assuming I'm not too far off - 9d08a5 is fix a definite user-surprise case - "I selected these things specifically, and somehow I'm not getting what I expect". Updating dest with latest, **might** be unexpected? Putting my user-hat on, I'd think the current behavior ('copy latest into dest') is what I generally would want, when copying, **except** in the "I've already decided on specific versions" case that's fixed by 9d08a5 .

I lean 60/40 towards  " 6dac5e would be More Surprising to me than the existing behavior (esp once 9d08a5 is in place)". I think I'd go w/ just the first commit.

My apologies for not responding to this sooner :(

Reply:
>My apologies for not responding to this sooner :(

No problem

> I lean 60/40 towards " 6dac5e would be More Surprising to me than the existing behavior (esp once 9d08a5 is in place)". I think I'd go w/ just the first commit.

So I don't necessarily know that it's more surprising, since depsolving-enabled copy would work roughly the same way (if you ignore that depsolving-enabled copy uses these "find_children" functions too, and only consider the way depsolving standard RPMs works).

But without a specific problem in mind we're probably better off sticking with the conservative change.